run/config.go

```diff
@@ -71,26 +70,26 @@ func ParseConfigFile(filepath string) (Config, base.LogSchema, ConfigStats, erro
 	}
 
 	var schema base.LogSchema
-	if s, err := checkAndCreateSchema(conf); err == nil {
-		schema = s
-	} else {
+	s, err := checkAndCreateSchema(conf)
+	if err != nil {
 		return conf, base.LogSchema{}, stats, err
 	}
+	schema = s
 
 	statsBuilder := NewConfigStatsBuilder(&schema)
 	statsBuilder.BeginTrackingFixedFields()
 
-	if err := bsupport.VerifyInputConfigs(conf.Inputs, schema, "inputs"); err != nil {
+	if err = bsupport.VerifyInputConfigs(conf.Inputs, schema, "inputs"); err != nil {
```

Comment:
doesn't this cause shadow or some warning about reusing variable? The idea of writing everything that way previously was to eliminate variable reusing.

Reply:
Nah, shadowing would be redeclaring a new `err` in an inner block/scope (an `if`, a loop or something like that), which is something that the linter will complain about (if we enable it, cause strict shadowing checks are disabled by default due to being "noisy").
But here it's just variable reusing, which is fine and happens all the time, since errors are usually supposed to be processed immediately after assignment.